### PR TITLE
ROC-2485: blocked API from accepting response when CCJ was requested

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-2485

### Change description ###
API blocked for saving response when response was already submitted for a claim or CCJ was requested.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseService.java [hunk 1]
```diff
@@ -59,4 +66,7 @@ private boolean isResponseAlreadySubmitted(final Claim claim) {
         return null != claim.getRespondedAt();
     }
 
+    private boolean isCCJAlreadyRequested(final Claim claim) {
+        return null != claim.getCountyCourtJudgmentRequestedAt();
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseService.java]: this reads wrong to me,
I would normally read this as:
```
return claim.getCountyCourtJudgmentRequestedAt() != null;
```

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseServiceTest.java [hunk 2]
```diff
@@ -68,11 +70,28 @@ public void saveShouldFinishSuccessfully() {
...
-        responseService.save(CLAIM_ID, DEFENDANT_ID, app, AUTHORISATION);
+        responseService.save(CLAIM_ID, DEFENDANT_ID, VALID_APP, AUTHORISATION);
 
         //then
         verify(eventProducer, once())
-            .createDefendantResponseEvent(eq(claim) );
+            .createDefendantResponseEvent(eq(claim));
+    }
+
+    @Test(expected = ForbiddenActionException.class)
```
Comment (geek-soft) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseServiceTest.java]: Would be better to subclass from `ForbiddenActionException` so then we can check specific exception. If method throws the same exception for many reasons that kind of test does not quite prove the right exception is thrown. Alternatively asserting error message would be some sort of solution.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseService.java [hunk 3]
```diff
@@ -44,9 +44,16 @@ public Claim save(
             throw new ForbiddenActionException("Response for the claim " + claimId + " was already submitted");
         }
 
+        if (isCCJAlreadyRequested(claim)) {
+            throw new ForbiddenActionException(
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseService.java]: We could have more specific  `CCJAlreadyRequestedException ` exception subclassing from `ForbiddedActionException`.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseServiceTest.java (hunk 2 above)
Comment (yarpo) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseServiceTest.java]: ok

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseService.java (hunk 3 above)
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/DefendantResponseService.java]: ok

**Pull request merged**